.github/workflows/python-app.yml

```diff
@@ -21,13 +24,35 @@ jobs:
 
     - name: Install dependencies and build project
       run: |
-        pip install prompt-toolkit
-        pip install jsonschema
-        pip install pytest
-        pip install pytest-cov
-        pip install -e .
-
-    - name: Run tests
-      run: pytest --cov=src/Models
+          pip install prompt-toolkit
+          pip install jsonschema
+          pip install pytest
+          pip install pytest-cov
+          pip install -e .
+        
+    - name: Install xmlstarlet
+      run: brew install xmlstarlet
+
+    - name: Get current and previous coverage
+      run: |
+          pytest --cov=src/Models --cov-report xml:current.xml
+          git checkout HEAD^
+          pytest --cov=src/Models --cov-report xml:previous.xml
```

Comment:
looks like you're getting a coverage report in an xml. Can you print that out or log it somewhere so that we can observe it?

Reply:
The XML files aren't very useful if displayed, there is a compare coverage tab within the mac build which shows you the exact values of the current and previous coverage.